rust/lance-index/src/scalar/bitmap.rs

```diff
@@ -253,15 +255,76 @@ impl ScalarIndex for BitmapIndex {
     }
 
     async fn load(store: Arc<dyn IndexStore>) -> Result<Arc<Self>> {
+        println!("Loading bitmap index from");
         let page_lookup_file = store.open_index_file(BITMAP_LOOKUP_NAME).await?;
-        let serialized_lookup = page_lookup_file
-            .read_range(0..page_lookup_file.num_rows(), None)
-            .await?;
+        // let serialized_lookup = page_lookup_file
+        //     .read_range(0..page_lookup_file.num_rows(), None)
+        //     .await?;
+
+        // For very large indices, read in chunks
+        const MAX_ROWS_PER_CHUNK: usize = 1_000_000;
+        let total_rows = page_lookup_file.num_rows();
+        println!("Total rows: {}", total_rows);
+
+        let mut index_map: BTreeMap<OrderableScalarValue, RowIdTreeMap> = BTreeMap::new();
+        let mut null_map = RowIdTreeMap::default();
+        let mut value_type: Option<DataType> = None;
+        let mut index_map_size_bytes = 0;
+
+        // Read and process in chunks to avoid offset overflow
+        for start_row in (0..total_rows).step_by(MAX_ROWS_PER_CHUNK as usize) {
+            let end_row = (start_row + MAX_ROWS_PER_CHUNK).min(total_rows);
+            let chunk = page_lookup_file
+                .read_range(start_row..end_row, None)
+                .await?;
+
+            if chunk.num_rows() == 0 {
+                continue;
+            }
+
+            // Set value_type from first chunk if not already set
+            if value_type.is_none() {
+                value_type = Some(chunk.schema().field(0).data_type().clone());
+            }
+
+            let dict_keys = chunk.column(0);
+            let binary_bitmaps = chunk.column(1);
+            let bitmap_binary_array = binary_bitmaps
+                .as_any()
+                .downcast_ref::<BinaryArray>()
+                .unwrap();
+
+            for idx in 0..chunk.num_rows() {
+                let key = OrderableScalarValue(ScalarValue::try_from_array(dict_keys, idx)?);
+                let bitmap_bytes = bitmap_binary_array.value(idx);
+                let bitmap = RowIdTreeMap::deserialize_from(bitmap_bytes).unwrap();
+
+                index_map_size_bytes += key.deep_size_of();
+                index_map_size_bytes += bitmap_bytes.len();
+                if key.0.is_null() {
+                    null_map = bitmap;
+                } else {
+                    index_map.insert(key, bitmap);
+                }
+            }
+        }
+
+        // Handle empty index case
+        let final_value_type = value_type.unwrap_or_else(|| {
+            // CULPRIT HERE????
+            // Default to some type if completely empty
+            DataType::Utf8
```

Comment:
it passes the newly added test `RUST_BACKTRACE=1  cargo test scalar::bitmap::tests::test_big_value -- --exact --nocapture`
but in order to pass `LANCE_LOG=info RUST_BACKTRACE=1 maturin develop && ../.venv/bin/python -m pytest python/tests/test_scalar_index.py -k test_create_index_empty_dataset -vv -s`, I need to comment this line and uncomment L318, L260~262